controller/api/destination/watcher/ip_watcher_test.go

```diff
@@ -736,3 +737,94 @@ spec:
 		}
 	})
 }
+
+func TestIpWatcherGetPod(t *testing.T) {
```

Comment:
Should we have a test to exercise cases where `hostNetwork: true`? What about the cases where we hit conflicts before?

Reply:
A test case with `hostNetwork: true` just means that there is a `Status.HostIP` set. Since we already test that there isn't really anything that test would add.

And yes we would hit conflicts on the first two `GetPod` calls in the test, and should correctly hit a conflict on the 3rd.